lib/elixir/src/elixir_tokenizer.erl

```diff
@@ -807,6 +809,16 @@ is_unnecessary_quote([Part], #elixir_tokenizer{warn_on_unnecessary_quotes=true}
 is_unnecessary_quote(_Parts, _Scope) ->
   false.
 
+unsafe_to_atom(Part, Line, Column, #elixir_tokenizer{static_atoms_encoder=StaticAtomsEncoder} = Scope) when
+    is_function(StaticAtomsEncoder) ->
```

Comment:
We can specify the arity of the `StaticAtomsEncoder` function:

```suggestion
    is_function(StaticAtomsEncoder, 2) ->
```

Reply:
Since this is called during the operation and not during input validation, I think it should stay as is, otherwise you can pass a function with arity 1 and never know why it was not called.

Reply:
🤔  IMO, such case should be treated as a different clause but that would not be terse and would be a bit too defensive 😅

Reply:
Oh good catch @josevalim, this would only make the clause fail and cascade. Agreed, better to have the actual call fail with wrong number of arguments then.